tests/storage-metadata

```diff
@@ -0,0 +1,342 @@
+
+        echo "==> Check the backup index file"
+        cat "${tmpdir}/backup/index.yaml"
+
+        if [ "${vers}" = "5.21" ] || [ "${vers}" = "5.0" ]; then
+            check_format_version1 "${tmpdir}/backup/index.yaml" instance
+
+            # The old format should be allowed for import in any case.
+            echo "==> Check the ${type} can be imported on LXD ${LXD_SNAP_CHANNEL}"
+            lxc import "${tmpdir}/i1.tar.gz" i1
+            [ "$(lxc query "/1.0/instances/i1?recursion=1" | jq '.snapshots | length')" = "1" ]
+
+            lxc delete -f i1
+        else
+            check_format_version2 "${tmpdir}/backup/index.yaml" instance
+
+            # Only LXD >= 5.21 allows importing the new format.
+            if hasNeededAPIExtension backup_metadata_version; then
+                echo "==> Check the ${type} can be imported on LXD ${LXD_SNAP_CHANNEL}"
+                lxc import "${tmpdir}/i1.tar.gz" i1
+                [ "$(lxc query "/1.0/instances/i1?recursion=1" | jq '.snapshots | length')" = "1" ]
+
+                lxc delete -f i1
+            fi
+        fi
+
+        rm -rf "${tmpdir}/backup"
+        rm "${tmpdir}/i1.tar.gz"
+
+        # Only LXD >= 5.21 allows setting the export version.
+        # We check for the backup_metadata_version API extension on both source (c1) and target (present) LXD.
+        # As both the server and client get shipped together with the snap, by testing for backup_metadata_version we can check if the client supports the --export-version flag.
+        if lxc exec c1 -- sh -c "$(declare -f hasNeededAPIExtension); hasNeededAPIExtension backup_metadata_version" && hasNeededAPIExtension backup_metadata_version; then
+            echo "==> Export using the old backup format 1 explicitly"
+            lxc export c1:i1 "${tmpdir}/i1.tar.gz" --export-version 1
+            tar -xzf "${tmpdir}/i1.tar.gz" -C "${tmpdir}" backup/index.yaml
+
+            echo "==> Check the backup index file"
+            cat "${tmpdir}/backup/index.yaml"
+            check_format_version1 "${tmpdir}/backup/index.yaml" instance
+
+            rm -rf "${tmpdir}/backup"
+
+            echo "==> Check the ${type} can be imported on LXD ${LXD_SNAP_CHANNEL}"
+            lxc import "${tmpdir}/i1.tar.gz" i1
+            [ "$(lxc query "/1.0/instances/i1?recursion=1" | jq '.snapshots | length')" = "1" ]
+
+            rm "${tmpdir}/i1.tar.gz"
+            lxc delete -f i1
+
+            echo "==> Export using the new backup format 2 explicitly"
+            lxc export c1:i1 "${tmpdir}/i1.tar.gz" --export-version 2
+            tar -xzf "${tmpdir}/i1.tar.gz" -C "${tmpdir}" backup/index.yaml
+
+            echo "==> Check the backup index file"
+            cat "${tmpdir}/backup/index.yaml"
+            check_format_version2 "${tmpdir}/backup/index.yaml" instance
+
+            rm -rf "${tmpdir}/backup"
+
+            # Only LXD >= 5.21 allows importing the new format 2.
```

Comment:
wont this comment be out of date once we backport support for v2 format to 5.21?